src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/world/FrozenTreasuresHighlighter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2022 NotEnoughUpdates contributors
+ *
+ * This file is part of NotEnoughUpdates.
+ *
+ * NotEnoughUpdates is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.miscfeatures.world;
+
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
+import io.github.moulberry.notenoughupdates.util.SBInfo;
+import io.github.moulberry.notenoughupdates.util.SpecialColour;
+import net.minecraft.block.Block;
+import net.minecraft.client.Minecraft;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.item.EntityArmorStand;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+import java.util.List;
+
+public class FrozenTreasuresHighlighter extends GenericBlockHighlighter {
+
+	private static final FrozenTreasuresHighlighter INSTANCE = new FrozenTreasuresHighlighter();
+
+	public static FrozenTreasuresHighlighter getInstance() {return INSTANCE;}
+
+
+	@Override
+	protected boolean isEnabled() {
+		return SBInfo.getInstance().getScoreboardLocation().equals("Glacial Cave")
+			&& NotEnoughUpdates.INSTANCE.config.world.highlightFrozenTreasures;
+	}
+
+	@Override
+	protected boolean isValidHighlightSpot(BlockPos key) {
+		World w = Minecraft.getMinecraft().theWorld;
+		if (w == null) return false;
+		Block b = w.getBlockState(key).getBlock();
+		return b == Blocks.ice;
+	}
+
+	@SubscribeEvent
+	public void onTickNew(TickEvent.ClientTickEvent event) {
+		if (event.phase != TickEvent.Phase.END||!isEnabled()) return;
```

Comment:
Please reformat the file. (CONTROL+ALT+L in IntellIJ)

Reply:
I didn't know about this shortcut! Thanks!